src/core/client/stream/tabs/Comments/Stream/AddACommentButton.tsx

```diff
@@ -1,32 +1,80 @@
 import React, { FunctionComponent, useCallback } from "react";
+import { graphql } from "react-relay";
 
 import { useCoralContext } from "coral-framework/lib/bootstrap";
+import { useLocal } from "coral-framework/lib/relay";
+import { NUM_INITIAL_COMMENTS } from "coral-stream/constants";
 import { POST_COMMENT_FORM_ID } from "coral-stream/constants";
 import { Flex, Icon } from "coral-ui/components/v2";
 import { Button } from "coral-ui/components/v3";
 import { useShadowRootOrDocument } from "coral-ui/encapsulation";
-import getElementWindowTopOffset from "coral-ui/helpers/getElementWindowTopOffset";
+
+import { AddACommentButtonLocal } from "coral-stream/__generated__/AddACommentButtonLocal.graphql";
 
 import styles from "./AddACommentButton.css";
 
 interface Props {
+  currentScrollRef: any;
   isQA?: boolean;
 }
-
-const AddACommentButton: FunctionComponent<Props> = ({ isQA = false }) => {
+const AddACommentButton: FunctionComponent<Props> = ({
+  isQA = false,
+  currentScrollRef,
+}) => {
   const { renderWindow } = useCoralContext();
+  const [{ totalCommentsLength }, setLocal] = useLocal<AddACommentButtonLocal>(
+    graphql`
+      fragment AddACommentButtonLocal on Local {
+        addACommentButtonClicked
+        totalCommentsLength
+        zKeyClickedLoadAll
+      }
+    `
+  );
   const root = useShadowRootOrDocument();
   const onClick = useCallback(() => {
     if (!renderWindow) {
       return;
     }
     const postCommentForm = root.getElementById(POST_COMMENT_FORM_ID);
     if (postCommentForm) {
-      renderWindow.scrollTo({
-        top: getElementWindowTopOffset(renderWindow, postCommentForm),
-      });
+      // Make sure that the Load all comments button is displayed even if it
+      // was previously clicked and no longer displayed.
+      // Then scroll to last comment, which is right above the Add a comment box,
+      // and scroll the post comment form into view.
+      setLocal({ addACommentButtonClicked: true });
+      setLocal({ zKeyClickedLoadAll: false });
+      if (totalCommentsLength) {
+        const indexToScroll =
+          totalCommentsLength < NUM_INITIAL_COMMENTS
+            ? totalCommentsLength - 1
+            : NUM_INITIAL_COMMENTS - 1;
+        if (currentScrollRef.current) {
+          currentScrollRef.current.scrollIntoView({
+            align: "center",
+            index: indexToScroll,
+            behavior: "auto",
+            done: () => {
+              let count = 0;
+              const stopExists = setInterval(async () => {
+                count += 1;
+                const stopElement = root.querySelector(
+                  `[data-index="${indexToScroll}"]`
+                );
+                if (stopElement !== undefined && stopElement !== null) {
+                  clearInterval(stopExists);
+                  postCommentForm.scrollIntoView();
+                }
+                if (count > 10) {
+                  clearInterval(stopExists);
+                }
```

Comment:
I see what you mean by the non-ideal retry interval. That being said, this is a clever and nicely contained solution that doesn't add a lot of complexity or issues anywhere but here in this snippet of code. Well done!